### PR TITLE
Add donation backup

### PR DESCRIPTION
**Pull request by MohammadPCh**
- #3549

### CODE REVIEW

File: pages/api/donation-backup.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+				body: JSON.stringify({
+					collection: process.env.MONGO_DONATION_COLLECTION,
+					database: process.env.MONGO_DONATION_DATABASE,
+					dataSource: process.env.MONGO_DONATION_DATA_SOURCE,
+					document: body,
+				}),
+			})
+				.then(response => response.json())
+				.then(data => console.log(data))
+				.catch(error => console.error('Error:', error));
```
Comment (RamRamez) [inline pages/api/donation-backup.js:21]: Is it possible to add sentry here to capture error?

**Review by RamRamez — CHANGES_REQUESTED** [review]

**Comment by RamRamez** [comment]
Thanks @MohammadPCh for this PR. I have a question. Where is the Mongo server? is it on the same server as impact-graph?

File: pages/api/donation-backup.js (hunk 1 above)
Comment (MohammadPCh) [reply pages/api/donation-backup.js:21]: Having issues with both servers simultaneously is uncommon, and in this scenario, there's little we can do. The MongoDB server, being highly stable and hosted by AWS, limits our intervention options. While I will make an addition, its effectiveness might be limited.

File: src/hooks/useCreateDonation.tsx [hunk 2]
```diff
@@ -153,6 +154,18 @@ export const useCreateDonation = () => {
...
+					chainId: transaction?.chainId!,
+					txHash: transaction?.hash,
+					amount: amount,
+					token,
+					projectId,
+					anonymous,
+					nonce: transaction?.nonce,
+					chainvineReferred,
+					walletAddress: transaction?.from,
+					symbol: token.symbol,
```
Comment (aminlatifi) [inline src/hooks/useCreateDonation.tsx:167]: Is it possible to include some information of the authentication token? e.g. expiration data, nonce, jwt, ...

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/hooks/useCreateDonation.tsx (hunk 2 above)
Comment (MohammadPCh) [reply src/hooks/useCreateDonation.tsx:167]: @aminlatifi I can add the user's token.

**Review by aminlatifi — APPROVED** [review]

**Comment by aminlatifi** [comment]
Thanks @MohammadPCh

**Review by RamRamez — APPROVED** [review]
LGTM, thanks @MohammadPCh

**Pull request merged**